### PR TITLE
Rover throttling

### PR DESCRIPTION
**Pull request by leogrim**
@stkem

### CODE REVIEW

File: kifi-backend/modules/rover/app/com/keepit/rover/model/ArticleInfoRepo.scala [hunk 1]
```diff
@@ -164,6 +166,12 @@ class ArticleInfoRepoImpl @Inject() (
       }
     }
   }
+
+  def setDomains(p: Int, s: Int)(implicit session: RWSession): Int = {
+    page(p, s).map { info =>
+      (for (r <- rows if r.id === info.id) yield r.domain).update(URI.parse(info.url).toOption.flatMap(_.host).map(_.name))
+    }.sum
+  }
```
Comment (yingjieMiao) [inline kifi-backend/modules/rover/app/com/keepit/rover/model/ArticleInfoRepo.scala:174]: `sum` exception when page is empty.

Comment (leogrim) [inline kifi-backend/modules/rover/app/com/keepit/rover/model/ArticleInfoRepo.scala:174]: I think they fixed that (as far as the REPL has been recently saying :)

Comment (yingjieMiao) [inline kifi-backend/modules/rover/app/com/keepit/rover/model/ArticleInfoRepo.scala:174]: ah, nice.

**Comment by stkem** [comment]
:+1:

**Pull request merged**